packages/convertkit/src/forms/index.tsx

```diff
@@ -1,69 +1,91 @@
 import * as React from 'react'
 
-//TODO: separate the styling and the behavior so that it can be customized
-
 const FORM_ID = process.env.NEXT_PUBLIC_CONVERTKIT_SIGNUP_FORM
 
 type ConvertkitSubscribeFormProps = {
   onSubmit?: () => void
+  button?: React.DetailedHTMLProps<
+    React.ButtonHTMLAttributes<HTMLButtonElement>,
+    HTMLButtonElement
+  >
+  /** text below submit button */
+  comment?: string
+  classNames?: {
+    input?: string
+    label?: string
+    button?: string
+    /** classNames for text below submit button */
+    comment?: string
+    /** children container classNames */
+    children?: string
+    form?: string
+    /** required field indicator (*) classNames */
+    asterisk?: string
+    buttonContainer?: string
+  }
 }
```

Comment:
I went with this `classNames` object approach, but am open to other suggestions. Seems simple and flexible enough, plus type descriptions make it nice to work with. Button component is replaceable since it's required at some times.

Reply:
i was immediately confused by the. among because of the [classnames library](https://github.com/JedWatson/classnames). 

is it typed? that’d be handy for using it. 

also think you could merge the defaults so they’d be useful individually and not require the entire object. 

the other approach that springs to mind is like chakra or other similar system where style props are individually addressable, which may provide a “nicer” api

i’d be very interested in seeing other existing example systems that approach this problem if they are available.

Reply:
Yeah not set on the naming either and will look if classnames library if that can be any help here. I'll also look for some real world examples and post my findings here.

> also think you could merge the defaults so they’d be useful individually and not require the entire object.

Merging the defaults is tricky since you can't change the precedence in html with the order of classnames. What matters is the specificity of css selectors. Gets weird quickly. And if I understand your point correctly, you don't need to specify the entire object and can change styles granularly `classNames: {{ button: "text-rose-500" }}`

Reply:
you’d merge the defaults and incoming object in javascript, not html so an incomplete incoming style object wouldn’t be incomplete and would still have default styles applied for any missing incoming properties

i’d like to see some “prior art” if possible because this feels like a problem that probably gets solved quite often